docs/architecture/adr-002-docs-structure.md

```diff
@@ -0,0 +1,89 @@
+# ADR 002: SDK Documentation Structure
+
+## Context
+
+There is a need for a scalable structure of the SDK documentation. Current documentation includes a lot of non-related SDK material, is difficult to maintain and hard to follow as a user. 
+
+Ideally, we would have:
+- All docs related to dev frameworks or tools live in their respective github repos (sdk repo would contain sdk docs, hub repo would contain hub docs, lotion repo would contain lotion docs, etc.)
+- All other docs (faqs, whitepaper, high-level material about Cosmos) would live on the website.
+
+## Decision
+
+Re-structure the `/docs` folder of the SDK github repo as follows:
+
+```
+docs/
+├── README/
+├── /gaia
+│   ├── overview.md
+│   ├── install.md
+│   ├── join-testnet.md
+│   ├── validator-node.md
+│   ├── validator-faq.md
+│   └── delegator-faq.md    
+├── reference/
+│   ├── baseapp
+│   ├── types
+│   ├── store
+│   ├── server
+│   ├── modules/
+│   │   ├── keeper
+│   │   ├── handler
+│   │   ├── cli
+│   ├── gas
+│   └── commands
+├── examples/
+│   ├── basecoin/
+│   └── democoin/
+├── clients
+│   ├── lite/
+│   ├── service-providers
+├── modules/
+└── architecture/
+```
+
+The files in each sub-folders do not matter and will likely change. What matters is the sectioning:
+
+- `README`: Landing page of the docs. Goal is to have a short explainer of the SDK and then channel people to the resource they need. The [sdk-tutorial](https://github.com/cosmos/sdk-application-tutorial/pulls) will be highlighted, as well as the `godocs`.
+- `gaia`: Contains all docs related to the `gaia` application. Will later be renamed to `cosmos-hub` or `chub` and probably moved to its own repository.
+- `reference`: Contains high-level explanations of the abstractions of the SDK. It does not contain specific code implementation and does not need to be updated often. **It is not an API specification of the interfaces**. API spec is the `godoc`. 
+- `examples`: Contain a couple examples of sdk application liker `bascoin` and `democoin`. Developers need to maintain them up-to-date and make sure they compiles as the SDK gets upgraded.
+- `clients`: Contains specs and infos about the various SDK clients.
+- `modules`: Contains specs of modules.
+- `architecture`: Contains architecture-related docs like the present one.
+
+Website docs will only include the following sections:
+- Intro (`README`)
+- `gaia`
+- `reference`
+- `clients`
+
+`architecture` and `examples` need not be displayed on the website. As for `modules`, we might need to think about creating a modules manager, but this is out of scope for this document.
```

Comment:
the files get rendered by virtue of being `.md` files, so they're accesible by default, even if not linked in the sidebar

Reply:
sure, this is what matters (that they're not in the sidebar)

Reply:
Agree with @gamarin2 here